.github/workflows/all.yml

```diff
@@ -0,0 +1,118 @@
+name: CI
+on:
+  # Enable manually triggering this workflow via the API or web UI
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+
+  lint:
+    runs-on: ubuntu-latest
+    env:
+      GOLANG_CI_VERSION: v1.49.0
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18.x
+      - name: golangci-lint on Go
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: ${{ env.GOLANG_CI_VERSION }}
+
+  test-prev:
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.18.x]
```

Comment:
Since you already use a matrix, I'd recommend using the only job for the tests, which could contain all go versions that we'd like to check (even the `tip`), like:
```yml
go-version: [1.18.x, 1.19.x, 1.20.x, tip]
```

The `Install Go` step will be a bit more complicated (there should be conditions for the tip and no-tip installation), Something like:

```
- name: Install Go ${{ matrix.go-version }}
  uses: actions/setup-go@v3
  with:
     go-version: ${{ matrix.go-version }}
- name: Install Go stable
   if: matrix.go-version == 'tip'
   uses: actions/setup-go@v3
   with:
      go-version: 1.x
- name: Install Go tip
   shell: bash
   if: matrix.go-version == 'tip'
   run: |
       go install golang.org/dl/gotip@latest
       gotip download
       echo "GOROOT=$HOME/sdk/gotip" >> "$GITHUB_ENV"
       echo "GOPATH=$HOME/go" >> "$GITHUB_ENV"
       echo "$HOME/go/bin" >> "$GITHUB_PATH"
       echo "$HOME/sdk/gotip/bin" >> "$GITHUB_PATH"
```

But as a result, the primary step with the running of the tests will be consolidated.

Reply:
Good idea! Thank you!